### PR TITLE
First pass for training loop and config loading

### PR DESCRIPTION
**Pull request by adrianSRoman**
TODO
- [x] Add trainer
- [x] Load config
- [x] Define data-loading process
- [x] Define loss function
- [x] Define dataset splitting
- [x] Enable logging
- [x] Integrate training board 
---
Finally:
- [x] Make sure to train and overfit (loss should go close to zero ideally)
- [x] Update readme with documentation on usage for all of the above

### CODE REVIEW

**Comment by adrianSRoman** [comment]
Overfitting

```
============== 5 epoch ==============
[0 seconds] Begin training...
Loss  0.00943161640316248
[2 seconds] End this epoch.
============== 6 epoch ==============
[0 seconds] Begin training...
Loss  0.008513818494975567
[2 seconds] End this epoch.
============== 7 epoch ==============
[0 seconds] Begin training...
Loss  0.008073325268924236
[2 seconds] End this epoch.
============== 8 epoch ==============
[0 seconds] Begin training...
Loss  0.007567078340798616
```

...

```
============== 89 epoch ==============
[0 seconds] Begin training...
Loss  8.968675683718175e-05
[2 seconds] End this epoch.
============== 90 epoch ==============
[0 seconds] Begin training...
Loss  0.000266181246843189
[2 seconds] End this epoch.
============== 91 epoch ==============
[0 seconds] Begin training...
Loss  8.054362842813134e-05
[2 seconds] End this epoch.
============== 92 epoch ==============
```
